anitya/lib/versions/rpm.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2017  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions
+# of the GNU General Public License v.2, or (at your option) any later
+# version.  This program is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.  You
+# should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# Any Red Hat trademarks that are incorporated in the source
+# code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission
+# of Red Hat, Inc.
+"""
+This adds support for comparing project versions using the RPM version rules.
+"""
+from __future__ import unicode_literals
+
+import functools
+import re
+
+from .base import Version
+
+
+try:
+    from rpm import labelCompare as _compare_rpm_labels
+except ImportError:
+    # Emulate RPM field comparisons as described in
+    # http://stackoverflow.com/questions/3206319/how-do-i-compare-rpm-versions-in-python/3206477#3206477
```

Comment:
It does have a license by default:
````
You agree that all Subscriber Content that You contribute to the Network is perpetually and irrevocably licensed to Stack Exchange under the Creative Commons Attribution Share Alike license. 
````
source: https://stackexchange.com/legal

And CC-BY-SA is not compatible with the GPL license making taking code from SO a little problematic :(